core/src/main/java/io/hyperfoil/core/generators/RandomCsvRowStep.java

```diff
@@ -0,0 +1,146 @@
+    * @param cummulativeProbs
+    * @param rowsList processed declarative data
+    * @param columns currently active columns
+    * @param var declarative data
+    */
+   public RandomCsvRowStep(String[][] rows, List<String> vars) {
+      this.rows = rows;
+      this.columnVars = vars.stream().map(SessionFactory::access).toArray(Access[]::new);
+   }
+
+   @Override
+   public boolean invoke(Session session) {
+      // columns provided by csv
+      ThreadLocalRandom random = ThreadLocalRandom.current();
+      int next = random.nextInt(rows.length);
+      int last = columnVars.length;
+      for (int i = 0, j = 0; i < last; i += 1) {
+         if (rows[next][i] != null) {
+            columnVars[j++].setObject(session, rows[next][i]);
+         }
+      }
+      return true;
+   }
+
+   @Override
+   public void reserve(Session session) {
+      Arrays.asList(columnVars).forEach(var -> var.declareObject(session));
+   }
+
+   public static class Builder extends BaseStepBuilder {
+      private String file;
+      private Boolean isHeaderRemoved;
+      private Boolean isQuotesRemoved;
+      private Map<String, Integer> builderColumns = new HashMap<>();
+      private int maxSize = 0;
+
+      public Builder(BaseSequenceBuilder parent) {
+         super(parent);
+      }
+
+      @Override
+      public List<Step> build(SerializableSupplier<Sequence> sequence) {
+         File f = new File(file);
+         if ( !f.exists() ) {
+            throw new BenchmarkDefinitionException("Supplied file cannot be found on system");
+         }
+         List<String[]> rows;
+         try ( BufferedReader reader = Files.newBufferedReader(Paths.get(f.getAbsolutePath())) ){
+            Predicate<String> comments = s -> (isHeaderRemoved ? !(s.trim().startsWith("#")) : true );
+            rows = reader.lines()
+                  .filter(comments)
+                  .map(s -> isQuotesRemoved ? s.replaceAll("\"","") : s)
+                  .map(line -> line.split(","))
+                  .collect(Collectors.toList());
+         } catch (IOException ioe) {
+            throw new BenchmarkDefinitionException(ioe.getMessage());
+         }
+         if (rows.isEmpty()) {
+            throw new BenchmarkDataException("Missing CSV row data. Rows were not detected after initial processing of file.");
+         }
+         rows.forEach(arr -> {for (int i = 0 ; i < arr.length; i += 1) {
+            if (!builderColumns.containsValue(i)) {
+               arr[i] = null;
+            }
+         }});
+         List<String> cols = new ArrayList<>();
+         cols.addAll(builderColumns.keySet());
+
+         return Collections.singletonList(new RandomCsvRowStep(rows.toArray(new String[][]{}), cols));
+      }
+
+      public ColumnsBuilder columns() {
+         return new ColumnsBuilder();
+      }
+
+      public Builder file(String file) {
+         this.file = file;
+         return this;
+      }
+
+      public Builder headers(String hasHeader) {
```

Comment:
Please make the parameter boolean. Parser will do the conversion automatically.

Reply:
Happy to.